### ISSUE TITLE
Feature request: inherit from collections.abc

### ISSUE THREAD

**Issue by cdonovick** [phanrahan/magma#178]
It would be nice if dict like objects inherited from `collections.abc.Mapping` /  `collections.abc.MutableMapping`  this would make them behave like dicts with minimal work.

**Comment by leonardt** [comment]
Makes sense, we need to come up with a list of places where this would be useful.

Some that I can think of
- [ ] `instance.interface`
- [ ] `definition.interface`
